### PR TITLE
Update CI deploy job

### PR DESCRIPTION
**Pull request by tylerjw**
### Description

Reference: https://github.com/ros-planning/moveit_tutorials/commit/4bcd8cca58a56316c7096ee9be92b3b488ffcbf5

@rhaschke would you mind reviewing this change?

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -10,51 +10,57 @@ on:
...
-      if: ${{ success() }}
-      uses: peaceiris/actions-gh-pages@v3
+
+    - name: Upload pages artifact
+      uses: actions/upload-pages-artifact@v1
+      if: github.repository_owner == 'ros-planning'
       with:
-        github_token: ${{ secrets.GITHUB_TOKEN }}
-        publish_dir: ./build/html
-        cname: moveit.picknik.ai
```
Comment (henningkayser) [inline .github/workflows/deploy.yml:60]: don't we still have to set the cname somewhere?

**Review by henningkayser — COMMENTED** [review]
Looks correct to me. Is this only a simplification or does it apply any non-obvious changes that we should document?

File: .github/workflows/deploy.yml (hunk 1 above)
Comment (tylerjw) [reply .github/workflows/deploy.yml:60]: I just pushed a fix for this.

**Review by henningkayser — APPROVED** [review]

File: .github/workflows/deploy.yml [hunk 2]
```diff
@@ -10,51 +10,60 @@ on:
...
+        path: build/html
+
+  deploy:
+    if: github.repository_owner == 'ros-planning' && github.ref == 'refs/heads/master'
```
Comment (henningkayser) [inline .github/workflows/deploy.yml:60]: @tylerjw this should be `refs/heads/main`, correct?

**Pull request merged**